### ISSUE TITLE
Outdated r.js / uglify.js, minifying jQuery 1.8 fails

### ISSUE THREAD

**Issue by molily** [jwhitley/requirejs-rails#64]
I’ve got a strange minification error when trying to use jQuery 1.8. Minimal setup:

application.js.coffee:

```
require ['jquery'], ->
```

requirejs.yml:

```
paths:
  jquery: 'jquery-1.8.0'
```

With unminified jquery-1.8.0.js in vendor/assets/javascripts.

When I compile the assets, the jQuery code isn’t minified correctly. There’s a named function expression in jQuery:

```
(function add( args ) {
    jQuery.each( args, function( _, arg ) {
        if ( jQuery.isFunction( arg ) && ( !options.unique || !self.has( arg ) ) ) {
            list.push( arg );
        } else if ( arg && arg.length ) {
            // Inspect recursively
            add( arg );
        }
    });
})( arguments );
```

The built-in r.js with the built-in UglifyJS compiles this to:

```
(function e(args){jQuery.each(args,function(_,arg){jQuery.isFunction(arg)&&(!options.unique||!self.has(arg))?list.push(arg):arg&&arg.length&&add(arg)})})(arguments)
```

The problem is the mismatch between `e` and `add`. `e` is the mangled function name, but it’s called using original name `add`. This throws an exception.

The bug does not appear when
- Building a sample project with r.js 2.0.6 on the command line
- optimize: 'none' in requirejs.yml, then minifying the output manually with uglifyjs 1.3.3 on the command line
- Replacing bin/r.js in the gem code with r.js 2.0.6

In short, I think the cause of the problem is just the outdated r.js with comes with requirejs-rails. Would you mind to update it?

Until then I’m just using jQuery 1.7.2 again. jQuery doesn’t have a named function expression there.

Thanks!
